### PR TITLE
Indirect - Update F(Q) fit functions and documentation

### LINKED ISSUE #24417: Indirect - Fit Functions and their documentation need updating
**Original reporter:** Spencer

### Expected behavior
FickDiff, Teixeirawater & HallRoss need correct units like Chudley-Elliott.

The documentation of the equations for Chudley-Elliott and HallRoss needs updating (and possibly others).

### Platforms affected
All

### PR DESCRIPTION
**Pull request by robertapplin**
**Description of work.**
This PR introduces updated fit functions and documentation for Hall Ross, Chudley-Elliot and Teixeira Water.

**To test:**
Read the updated documentation and check that the HWHM within the scripts match those within the documentation for Hall Ross and Chudley-Elliot.

Fixes #24417

#### Reviewer ####

Please comment on the following ([full description](http://developer.mantidproject.org/IndividualTicketTesting/)):

##### Code Review #####

- Is the code of an acceptable quality?
- Does the code conform to the [coding standards](http://developer.mantidproject.org/Standards/)?
- Are the unit tests small and test the class in isolation?
- If there are changes in the release notes then do they describe the changes appropriately?

##### Functional Tests #####

- Do changes function as described? Add comments below that describe the tests performed?
- Do the changes handle unexpected situations, e.g. bad input?
- Has the relevant (user and developer) documentation been added/updated?

Does everything look good? Mark the review as **Approve**. A member of `@mantidproject/gatekeepers` will take care of it.

### CODE REVIEW

File: Framework/PythonInterface/plugins/functions/ChudleyElliot.py [hunk 1]
```diff
@@ -2,15 +2,18 @@
 #
 # Copyright &copy; 2007 ISIS Rutherford Appleton Laboratory UKRI,
 #     NScD Oak Ridge National Laboratory, European Spallation Source
-#     & Institut Laue - Langevin
+#     & Institute Laue - Langevin
```
Comment (TTitcombe) [inline Framework/PythonInterface/plugins/functions/ChudleyElliot.py]: It's actually called `Institut`

File: Framework/PythonInterface/plugins/functions/ChudleyElliot.py [hunk 2]
```diff
@@ -32,12 +35,22 @@ def function1D(self, xvals):
...
                               / (xvals * length))/tau
-
         return hwhm
 
+    def functionDeriv1D(self, xvals, jacobian):
+        tau = self.getParameterValue("Tau")
+        length = self.getParameterValue("L")
+
+        for i, x in enumerate(xvals, start=0):
+            s = 1.0 - math.sin(x*length)/(x*length)
```
Comment (soininen) [inline Framework/PythonInterface/plugins/functions/ChudleyElliot.py]: `math.sin(x*length)/(x*length)` produces a singularity at `x = 0` which shouldn't be there. The correct function to use here would be `numpy.sinc()` (see [Wikipedia](https://en.wikipedia.org/wiki/Sinc_function) and [numpy docs](https://docs.scipy.org/doc/numpy-1.7.0/reference/generated/numpy.sinc.html#numpy.sinc)): `numpy.sinc(x*length/numpy.pi)`.

File: Framework/PythonInterface/plugins/functions/TeixeiraWater.py [hunk 3]
```diff
@@ -31,9 +33,21 @@ def init(self):
     def function1D(self, xvals):
         tau = self.getParameterValue("Tau")
         length = self.getParameterValue("L")
-        q2l2 = np.square(length * np.array(xvals))
-        return (self.hbar/tau) * q2l2 / (6 + q2l2)
+        xvals = np.array(xvals)
+
+        with np.errstate(divide='ignore'):
+            hwhm = self.hbar * xvals * xvals * length / (tau * (1 + xvals * xvals * length))
```
Comment (soininen) [inline Framework/PythonInterface/plugins/functions/TeixeiraWater.py]: Is this correct? In the documentation, `length` is also squared. And by the way, using `numpy.square()` might be considered here (and other parts of this PR as well).

**Review by soininen — CHANGES_REQUESTED** [review]
A few comments below. Should this have release notes as well?

File: Framework/PythonInterface/plugins/functions/TeixeiraWater.py (hunk 3 above)
Comment (soininen) [reply Framework/PythonInterface/plugins/functions/TeixeiraWater.py]: Also, the documentation says 6 + (QL)^2 instead of 1 + Q^2 * L

Comment (robertapplin) [reply Framework/PythonInterface/plugins/functions/TeixeiraWater.py]: @SpencerHowells is the documentation correct here?

Comment (robertapplin) [reply Framework/PythonInterface/plugins/functions/TeixeiraWater.py]: @soininen 6 + (QL)^2 is correct, thanks for spotting this!

**Comment by robertapplin** [comment]
@soininen this should be ready to go now

**Review by soininen — APPROVED** [review]

**Pull request merged**